COMPATIBILITY.md

```diff
@@ -0,0 +1,80 @@
+# Compatibility Guarantee
+## Introduction
+
+The release of InfluxDB version 1 is a major milestone in the
+development of the software lifecycle. InfluxDB 1 is a stable platform
+for the growth of programs and projects that utilize InfluxDB as a
+database.
+
+## Expectations
+
+Although we expect that the vast majority of users will maintain this
+compatibility over time, it is impossible to guarantee that no future
+change will break any program. This document is an attempt to set
+expectations for the compatibility of InfluxDB 1 in the future. There
+are a number of ways in which a client that works today may fail to do
+so after a future point release. They are unlikely but worth recording.
+
+* Security. A security issue in the specification or implementation may
+  come to light whose resolution requires breaking compatibility. We
+  reserve the right to address such security issues.
+* Unspecified behavior. The InfluxDB specification tries to be explicit
```

Comment:
Limiting what this guarantee covers to well defined pieces will make it much easier to ensure we are keeping the 1.0 promise. We could add a couple paragraphs to the introduction like this:

> InfluxDB 1 covers the following things:
>     - the specification of the [InfluxQL syntax]()
>     - the [line protocol]()
>     - the [HTTP API]()
>     - the [Go client library]()
>     - (Anything else? Configuration? On disk format?)
> The InfluxDB 1 release includes implementations of the above features in the form of the `influxd` binary and the InfluxDB client library found in the `/client` directory at the root of the project.

> It is intended that programs relying on the InfluxDB interfaces and behavior covered under the InfluxDB 1 specification will continue to run correctly, unchanged, over the lifetime of the product. At some indefinite point, an InfluxDB 2 may be released, but until that time, programs using InfluxDB that work today should continue to work even as future "point" releases of InfluxDB 1 arise (InfluxDB 1.1, InfluxDB 1.2, etc.).

The scope of the things listed above may grow, acquiring new features, but not in a way that breaks existing InfluxDB 1 interfaces and behavior.